src/main/java/org/craftercms/studio/impl/v1/service/security/SecurityServiceImpl.java

```diff
@@ -151,28 +168,47 @@ public PermissionsConfigTO execute() {
             }
         }
 
-        StudioCacheContext globalCacheContext = new StudioCacheContext("###GLOBAL###", true);
+        StudioCacheContext globalCacheContext = new StudioCacheContext(CStudioConstants.CACHE_GLOBAL_SCOPE, true);
         PermissionsConfigTO globalRolesConfig = cacheTemplate.getObject(globalCacheContext, new Callback<PermissionsConfigTO>() {
             @Override
             public PermissionsConfigTO execute() {
                 return loadGlobalRolesConfiguration();
             }
-        }, "###GLOBAL###", globalConfigPath, globalRoleMappingsFileName);
+        }, CStudioConstants.CACHE_GLOBAL_SCOPE, globalConfigPath, globalRoleMappingsFileName);
         PermissionsConfigTO globalPermissionsConfig = cacheTemplate.getObject(globalCacheContext, new Callback<PermissionsConfigTO>() {
             @Override
             public PermissionsConfigTO execute() {
                 return loadGlobalPermissionsConfiguration();
             }
-        }, "###GLOBAL###", globalConfigPath, globalPermissionsFileName);
+        }, CStudioConstants.CACHE_GLOBAL_SCOPE, globalConfigPath, globalPermissionsFileName);
         Set<String> roles = new HashSet<String>();
         addGlobalUserRoles(user, roles, globalRolesConfig);
         addGlobalGroupRoles(roles, groups, globalRolesConfig);
         permissions.addAll(populateUserGlobalPermissions(path, roles, globalPermissionsConfig));
         return permissions;
     }
 
-    protected void addGlobalUserRoles(String user, Set<String> roles, PermissionsConfigTO rolesConfig) {
-        Set<String> groups = securityProvider.getUserGroups(user);
+    protected void addGlobalUserRoles( final String user, Set<String> roles, PermissionsConfigTO
+            rolesConfig) {
+
+        StudioCacheContext cacheContext = new StudioCacheContext(CStudioConstants.CACHE_USERS_SCOPE, true);
+        CacheService cacheService = cacheTemplate.getCacheService();
+        generalLockService.lock(cacheContext.getId());
+        try {
+            if (!cacheService.hasScope(cacheContext)) {
```

Comment:
Please put the creation of the scope in an init/constructor and the removal of the scope in a destructor or similar and just use the cache without checking/rechecking/re-creating every time.